api/api.go

```diff
@@ -76,6 +76,14 @@ func (srv *Server) initAPI(addr string) {
 		handleHTTPRequest(mux, "/miner/submitblock", srv.minerSubmitblockHandler)
 		handleHTTPRequest(mux, "/miner/headerforwork", srv.minerHeaderforworkHandler)
 		handleHTTPRequest(mux, "/miner/submitheader", srv.minerSubmitheaderHandler)
+		handleHTTPRequest(mux, "/miner/connecttopool", srv.minerConnectToPoolHandler)
+		handleHTTPRequest(mux, "/miner/poolheaderforwork", srv.minerPoolHeaderForWorkHandler)
+		handleHTTPRequest(mux, "/miner/submitpoolheader", srv.minerSubmitPoolHeaderHandler)
```

Comment:
idempotency is only part of the picture. I think in general it's confusing to use PUT when you aren't actually PUTting anything.